api/lib/network-management.js

```diff
@@ -103,3 +105,33 @@ export const mapRequestToSubnet = (req) => {
 
   return addr
 }
+
+export const logNetworkInfo = async (pgClient, req, stationId, inetGroup) => {
+  // Update the station_id's network_info_update_history row if it's older than 10 minutes
+  const { rows } = await pgClient.query(`
+    INSERT INTO network_info_update_history (station_id, updated_at)
+    VALUES ($1, NOW())
+    ON CONFLICT (station_id) DO UPDATE
+    SET updated_at = NOW()
+    WHERE network_info.updated_at < NOW() - INTERVAL '10 minutes'
+    RETURNING station_id
+  `, [stationId])
+
+  // Don't record to InfluxDB if we didn't update the row
+  if (rows.length === 0) return
```

Comment:
Thoughts on this as a method of avoiding excessive InfluxDB writes?

Reply:
I don't think we need 10 minutes precision, recording once per day is good enough. And with 1d precision, I think we don't need to compare to only record when the network changed, we can just record always. Could you simplify the setup with this? I think we only need to track whether we already recorded for this station id in this day.

Since this query will be performed for every measurement recorded, performance is crucial.

In pseudo code, the implementation could be somewhat like this:

```js
const stationIDsSeen = []
setInterval(() => { stationIDsSeen.length = 0 }, ONE_DAY)
// ...
if (!stationIDsSeen.includes(stationID)) {
  stationIDsSeen.push(stationID)
  record()
}
```

The list of Station IDs should be small enough to fit in memory easily. The only downside of using in-memory is that after deploy every station will get recorded again. Therefore I think we still need to use the DB for this

Reply:
All makes sense to me, but your last sentence

> Therefore I think we still need to use the DB for this

Do you mean we you think we shouldn't use the in-memory list and instead go with the current change using a new SQL table, but change it to 24h?

Reply:
I was wondering if we could just use an in-memory structure - the participants list is small enough. But, the downside is that we will record all participants after a restart, so I think it's not an option.

I think the best solution is to use an SQL table, but instead of comparing dates on INSERT (expensive), periodically empty the table

Reply:
sorry, which part was too expensive?

Reply:
Sorry should have clarified, from your comment above:

> I think the best solution is to use an SQL table, but instead of comparing dates on INSERT (expensive), periodically empty the table

Reply:
@PatrickNercessian I discussed this with @juliangruber today. We decided it's best to radically simplify the first version and then iteratively improve it as needed.

Let's drop the entire `network_info_update_history` table & related code and keep the state in memory.

We consider the downside of having duplicate data points (caused by the in-memory state not shared by the four processes/service instances) to be acceptable.

We did a bit of napkin math: in the worst case, we have ~10 deployments per day, which means 40 duplicate submissions of the geolocation data. With 20k unique station instances, that means ~800k data points daily. We think that's not unreasonable, considering how many other different data points we are writing from other parts of the Station & Spark systems.

If this becomes a problem, we can add a caching layer then.

Could you please rework the pull request accordingly?

Also, remember to use a different bucket for these location data points; we will want to set a very short data retention period - something like 3 days.

Reply:
Are we sure? The only listed downside to the existing PR approach is that we can't easily modify the fact that it updates every 24h.

The in-memory state does not give us this functionality either, and adds the unnecessary duplicate submissions (even if they're not a real problem at the moment), and, to me, it does not seem obvious that it is much simpler.

Actually, as I'm thinking about it, I guess we could have the "updated_at" timestamp in-memory as well, and use JS logic rather than SQL logic to decide whether to update? Is this what you meant?

Just want to double check, if you're certain about the in-memory state I can go ahead and start implementation!

Reply:
We are concerned about the performance impact of making another SQL query every time we store a new measurement.

The idea is to remove all SQL from this pull request and keep the update history in memory.

If this becomes a problem, we would like to explore more options:
- share the update history via Redis
- query InfluxDB to find the last update time
- add in-memory cache for those InfluxDB requests
- store the update history in SQL as you proposed here
- maybe something else

The point is to ship the code collecting location data first, observe how it works in production, and only then fix what needs fixing.

Reply:
Ok! So were you thinking we do an "updated_at" timestamp (or an "last_update_day" date) in-memory, with JS logic?

Reply:
I think something like this could still work, as posted in the beginning of the thread:

```
const stationIDsSeen = []
setInterval(() => { stationIDsSeen.length = 0 }, ONE_DAY)
// ...
if (!stationIDsSeen.includes(stationID)) {
  stationIDsSeen.push(stationID)
  record()
}
```

Reply:
Got it! Addressed in https://github.com/filecoin-station/spark-api/pull/339/commits/66b9d3e08937267824c36c61f708c182230d187e